File: prozesse_und_threads/loesung/01_fork.py
#!/usr/bin/env python3

import sys
import os

if __name__ == "__main__":
    limit = 100
    pid = os.fork()
    if pid < 0:
        print("Fork failed")
        sys.exit(-1)
    elif pid == 0:
        print("begin child")
        for i in range(0, limit):
            print("Child:", i)
        print("end child")
        os._exit(0)
    else:
        print("begin parent")
        for j in range(0, limit):
            print("Parent:", j)
        print("end parent")
        sys.exit(0)
        

File: prozesse_und_threads/loesung/01_fork.cpp
#include <unistd.h>
#include <iostream>

using namespace std;

int main()
{
    int limit = 100;
    int counter = 0;
    pid_t pid = fork();

    if (pid == 0)
    {
        cout << "begin child" << endl;
        for (int i = 0; i < limit; i++)
        {
            cout << "Child: " << i << endl;
        }
        cout << "end child" << endl;
    }
    else if (pid > 0)
    {
        cout << "begin parent" << endl;
        for (int j = 0; j < limit; j++)
        {
            cout << "Parent: " << j << endl;
        }
        cout << "end parent" << endl;
    }
    else
    {
        cout << "Fork failed!" << endl;
        return -1;
    }
    return 0;
}